compiler/passes/src/common/graph/mod.rs

```diff
@@ -107,6 +107,17 @@ impl<N: Node> DiGraph<N> {
         Ok(finished)
     }
 
+    /// Keeps a subset of nodes, and removes the rest from the graph.
```

Comment:
Add detail on how nodes are retained based on whether or not they are sources or destination in an edge.